scraping/reddit/utils.py

```diff
@@ -37,6 +39,109 @@ def validate_reddit_content(
             content_size_bytes_validated=entity_to_validate.content_size_bytes,
         )
 
+    # Check Reddit id
+    if content_to_validate.id != actual_content.id:
+        bt.logging.info(
+            f"Reddit ids do not match: {actual_content} != {content_to_validate}"
+        )
+        return ValidationResult(
+            is_valid=False,
+            reason="Reddit ids do not match",
+            content_size_bytes_validated=entity_to_validate.content_size_bytes,
+        )
+
+    # Check Reddit url
+    if content_to_validate.url != actual_content.url:
+        bt.logging.info(
+            f"Reddit urls do not match: {actual_content} != {content_to_validate}"
+        )
+        return ValidationResult(
+            is_valid=False,
+            reason="Reddit urls do not match",
+            content_size_bytes_validated=entity_to_validate.content_size_bytes,
+        )
+
+    # Check Reddit username
+    if content_to_validate.username != actual_content.username:
+        bt.logging.info(
+            f"Reddit usernames do not match: {actual_content} != {content_to_validate}"
+        )
+        return ValidationResult(
+            is_valid=False,
+            reason="Reddit usernames do not match",
+            content_size_bytes_validated=entity_to_validate.content_size_bytes,
+        )
+
+    # Check Reddit community
+    if content_to_validate.community != actual_content.community:
+        bt.logging.info(
+            f"Reddit communities do not match: {actual_content} != {content_to_validate}"
+        )
+        return ValidationResult(
+            is_valid=False,
+            reason="Reddit communities do not match",
+            content_size_bytes_validated=entity_to_validate.content_size_bytes,
+        )
+
+    # Check Reddit body
+    if content_to_validate.body != actual_content.body:
+        bt.logging.info(
+            f"Reddit bodies do not match: {actual_content} != {content_to_validate}"
+        )
+        return ValidationResult(
+            is_valid=False,
+            reason="Reddit bodies do not match",
+            content_size_bytes_validated=entity_to_validate.content_size_bytes,
+        )
+
+    # Check Reddit created_at
+    # If checking an data entity with obfuscated content we compare to the entity directly instead.
+    if allow_obfuscated_content_date:
+        if entity_to_validate.datetime != actual_content.created_at:
```

Comment:
It's not obvious to me that this is correct (perhaps a more comprehensive comment would be helpful?)

In this case, why is it guaranteed that the entity has an obfuscated time? In fact, is it also guaranteed the created_at is obfuscated? My understanding was "actual_content" will always have full precision

Reply:
Fixed per discussion.